ksqldb-engine/src/test/java/io/confluent/ksql/planner/plan/PullFilterNodeTest.java

```diff
@@ -1099,13 +1172,80 @@ public void shouldThrowOnMultiKeyExpressions() {
             expression,
             metaStore,
             ksqlConfig,
-            false
+            false,
+            plannerOptions
         ));
 
     // Then:
     assertThat(e.getMessage(), containsString("An equality condition on the key column cannot be combined with other comparisons"));
   }
 
+  @SuppressWarnings("unchecked")
+  @Test
+  public void shouldExtractConstraintWithMultiKeyExpressions_tableScan() {
+    // Given:
+    when(plannerOptions.getTableScansEnabled()).thenReturn(true);
+    final Expression expression1 = new ComparisonExpression(
+        Type.EQUAL,
+        new UnqualifiedColumnReferenceExp(ColumnName.of("K")),
```

Comment:
Shouldn't this be K1 and K2 respectively?

Reply:
The name of the test is slightly misleading.  The test says that if you make multiple references in the same disjunct to a given key column, it must do a table scan.  This is a little strict and future optimizations will likely make this unnecessary.  An example:

`SELECT * FROM T1 WHERE K > 2 AND K = 5;` This could just be simplified as 
`SELECT * FROM T1 WHERE K = 5;` but we don't yet do this simplification and will instead consider too difficult to extract, resulting in a table scan.

Reply:
The test name makes it sound like a multi-column key.  I changed it to `shouldExtractConstraintWithMultipleKeyExpressions_tableScan`

Reply:
Ack, thanks!